### PR TITLE
Organize tests

### PR DESCRIPTION
**Pull request by kiran-chaudhary**
Moved tests to their respective unit tests file.

The tests cover the cases which validate error message when required args are not passed.

As per discussion in last sprint with @timabrmsn: 
Whether we need these tests needs to be rethought. This generally seems like we are testing click framework behaviour and what we intend to test is whether we have configured the required arguments properly or not.

So if we do not need these tests, only work to be done in the PR is remove the tests.

### CODE REVIEW

**Comment by unparalleled-js** [comment]
> Moved tests to their respective unit tests file.
> 
> The tests cover the cases which validate error message when required args are not passed.
> 
> As per discussion in last sprint with @timabrmsn:
> Whether we need these tests needs to be rethought. This generally seems like we are testing click framework behaviour and what we intend to test is whether we have configured the required arguments properly or not.
> 
> So if we do not need these tests, only work to be done in the PR is remove the tests.

I see what you are saying and am open to purer alternatives.  These tests _do_ prevent bugs however (remember the `delete profile` bug).

File: tests/cmds/test_departing_employee.py [hunk 1]
```diff
@@ -337,3 +338,29 @@ def test_remove_departing_employee_when_user_not_on_list_prints_expected_error(
...
     )
+
+
+@pytest.mark.parametrize(
+    "command, error_msg",
+    [
+        ("{} add".format(DEPARTING_EMPLOYEE_COMMAND), "Missing argument 'USERNAME'."),
+        (
+            "{} remove".format(DEPARTING_EMPLOYEE_COMMAND),
+            "Missing argument 'USERNAME'.",
```
Comment (unparalleled-js) [inline tests/cmds/test_departing_employee.py:349]: Opportunity to re-use error messages or have a helper method for creating them like

```
_get_err_opt_text("username")
_get_err_arg_text("CSV_FILE'")
```

`return "Missing option '--{}' .".format(opt)`

File: tests/cmds/test_alert_rules.py [hunk 2]
```diff
@@ -343,3 +345,35 @@ def test_remove_when_user_not_on_rule_raises_expected_error(runner, cli_state, m
...
+            "{} bulk add".format(ALERT_RULES_COMMAND),
+            "Error: Missing argument 'CSV_FILE'.",
+        ),
+        (
+            "{} bulk remove".format(ALERT_RULES_COMMAND),
+            "Error: Missing argument 'CSV_FILE'.",
+        ),
+    ],
+)
+def test_alert_rules_command_returns_error_exit_status_when_missing_required_parameters(
```
Comment (unparalleled-js) [inline tests/cmds/test_alert_rules.py:374]: Typically we put the condition aspect of the tests in the middle of the signature

`test_alert_rules_command_when_missing_required_parameters_errors`

Comment (unparalleled-js) [inline tests/cmds/test_alert_rules.py:374]: This test claims to be focused on the error status but also checks the error message, so replacing `returns_error_exit_status` with `errors` is more honest

**Comment by unparalleled-js** [comment]
> Moved tests to their respective unit tests file.
> 
> The tests cover the cases which validate error message when required args are not passed.
> 
> As per discussion in last sprint with @timabrmsn:
> Whether we need these tests needs to be rethought. This generally seems like we are testing click framework behaviour and what we intend to test is whether we have configured the required arguments properly or not.
> 
> So if we do not need these tests, only work to be done in the PR is remove the tests.

I see what you are saying and am open to purer alternatives.  This seems easiest thought.

**Comment by unparalleled-js** [comment]
Are there missing `cases` missing arg tests?

File: tests/cmds/test_alert_rules.py (hunk 2 above)
Comment (unparalleled-js) [reply tests/cmds/test_alert_rules.py:374]: This is also mentioned in the CONTRIBUTING guide

File: tests/cmds/test_departing_employee.py (hunk 1 above)
Comment (unparalleled-js) [reply tests/cmds/test_departing_employee.py:349]: This is not a mandatory suggestion!  Just something I noticed.

**Comment by kiran-chaudhary** [comment]
> > Moved tests to their respective unit tests file.
> > The tests cover the cases which validate error message when required args are not passed.
> > As per discussion in last sprint with @timabrmsn:
> > Whether we need these tests needs to be rethought. This generally seems like we are testing click framework behaviour and what we intend to test is whether we have configured the required arguments properly or not.
> > So if we do not need these tests, only work to be done in the PR is remove the tests.
> 
> I see what you are saying and am open to purer alternatives. This seems easiest thought.



> > Moved tests to their respective unit tests file.
> > The tests cover the cases which validate error message when required args are not passed.
> > As per discussion in last sprint with @timabrmsn:
> > Whether we need these tests needs to be rethought. This generally seems like we are testing click framework behaviour and what we intend to test is whether we have configured the required arguments properly or not.
> > So if we do not need these tests, only work to be done in the PR is remove the tests.
> 
> I see what you are saying and am open to purer alternatives. These tests _do_ prevent bugs however (remember the `delete profile` bug).

Yea, I gave the same example why we might need this. However I wanted to bring it forward to everyone as others where on leave.

**Review by unparalleled-js — APPROVED** [review]
It would be nice to make the test signatures more formal

**Pull request merged**